internal/cmd/commands/database/funcs.go

```diff
@@ -46,24 +46,29 @@ func migrateDatabase(ctx context.Context, ui cli.Ui, dialect, u string, initiali
 	// This is an advisory lock on the DB which is released when the DB session ends.
 	if err := man.ExclusiveLock(ctx); err != nil {
 		ui.Error("Unable to capture a lock on the database.")
+		_ = man.Close(ctx)
 		return noop, 2
 	}
 	unlock := func() {
 		// We don't report anything since this should resolve itself anyways.
 		_ = man.ExclusiveUnlock(ctx)
+		_ = man.Close(ctx)
 	}
 
 	st, err := man.CurrentState(ctx)
 	if err != nil {
 		ui.Error(fmt.Errorf("Error getting database state: %w", err).Error())
+		_ = man.Close(ctx)
```

Comment:
This still kind of has the same issue. We will call `Close` prior to `unlock`. If you look at where we happen to call `migrateDatabase`, we always do a `defer clean()` which is what the caller calls `unlock`. We do these even before checking for an error, so we will call `unlock` even in an error case. So just adding the `Close` to `unlock` should be sufficient given how it is used.

Reply:
🤦🏻 I didn't notice it was still returning `unlock`.